testing/cli/test_cli_sct_analyze_lesion.py

```diff
@@ -1,16 +1,71 @@
-import os
-
 import pytest
 import logging
 
-from spinalcordtoolbox.scripts import sct_analyze_lesion
+import pickle
+import numpy as np
+
+from spinalcordtoolbox.utils import sct_test_path, extract_fname
+from spinalcordtoolbox.scripts import sct_analyze_lesion, sct_label_utils
 
 logger = logging.getLogger(__name__)
 
 
+@pytest.fixture()
+def dummy_lesion(tmp_path):
+    """Define a fake voxel lesion using the specified dimensions."""
+    # TODO: Parametrize this fixture to test other lesion dimensions
+    starting_coord = [29, 27, 25]
+    dim = [1, 4, 1]
+
+    # Format the coordinates into a str argument that `-create` can accept
+    coordinates = []
+    for x in range(starting_coord[0], starting_coord[0] + dim[0]):
+        for y in range(starting_coord[1], starting_coord[1] + dim[1]):
+            for z in range(starting_coord[2], starting_coord[2] + dim[2]):
+                coord = [str(x), str(y), str(z), "1"]
+                coordinates.append(",".join(coord))
+    create_arg = ":".join(coordinates)
+
+    # Create the lesion mask file and output to a temporary directory
+    path_ref = sct_test_path("t2", "t2.nii.gz")
+    path_out = str(tmp_path/"lesion.nii.gz")
+    sct_label_utils.main(argv=['-i', path_ref, '-o', path_out,
+                               '-create', create_arg])
+
+    # Compute the expected measurements from the provided dimensions
+    measurements = {
+        # NB: 'sct_analyze_lesion' treats lesions as cylinders. So:
+        #   - Y axis: Length of the cylinder
+        'length [mm]': dim[1],
+        #   - X/Z plane: Cross-sectional slices of the cylinder.
+        #                Specifically, 'max_equivalent_diameter' takes the X/Z
+        #                cross-sectional area of the lesion (which is computed
+        #                using square voxels), then finds the diameter of an
+        #                equivalent *circle* with that same area:
+        #                   a = pi*r^2
+        #                -> a = pi*(d/2)^2
+        #                -> d = 2*sqrt(a/pi)
+        'max_equivalent_diameter [mm]': 2 * np.sqrt(dim[0] * dim[2] / np.pi),
+        'volume [mm3]': dim[0] * dim[1] * dim[2],
+    }
+
+    return path_out, measurements
+
+
 @pytest.mark.sct_testing
-@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
-def test_sct_analyze_lesion_output_file_exists():
+def test_sct_analyze_lesion_output_file_exists(dummy_lesion, tmp_path):
```

Comment:
Whoops. Forgot to rename the test.

```suggestion
def test_sct_analyze_lesion_results_with_dummy_data(dummy_lesion, tmp_path):
```